src/test/java/io/pinecone/PineconeConfigTest.java

```diff
@@ -0,0 +1,66 @@
+package io.pinecone;
+
+import io.pinecone.configs.PineconeConfig;
+import io.pinecone.exceptions.PineconeConfigurationException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class PineconeConfigTest {
```

Comment:
New test file focusing on `PineconeConfig` which has the `user-agent` generation logic for now.